build-scripts/build_compiler.js

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env node
+/*
+ * Copyright 2018 The Closure Compiler Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const {spawn, spawnSync} = require('child_process');
+const ncp = require('ncp');
+const Semver = require('semver');
+const version = require('../lerna.json').version;
+const fs = require('fs');
+const packageVer = new Semver(version);
+const glob = require('glob');
+
+const gitCmd = spawnSync('git', ['tag', '--points-at', 'HEAD'], {
+  cwd: './compiler'
+});
+let compilerVersion = '1.0-SNAPSHOT';
+if (gitCmd.status === 0) {
+  const currentTag = gitCmd.stdout.toString().replace(/\s/g, '');
+  let normalizedTag = currentTag;
+  if (normalizedTag) {
+    normalizedTag = currentTag.replace(/^([a-z]+-)?v\d{8}(.*)$/,
+        (match, g1, g2) => match.substr((g1 || '').length, match.length - (g1 || '').length - (g2 || '').length));
```

Comment:
I find this replacement function hard to follow, but I think you're just trying to pull out the middle part.
Why not just this?

```javascript
// e.g. "abc-v20181022otherjunk" -> "v20181022"
normalizedTag = currentTag.replace(/^([a-z]+-)?(v\d{8}).*$/, "$2");
```